sdk/nodejs/docker.ts

```diff
@@ -269,9 +269,6 @@ async function buildAndPushImageWorkerAsync(
 
     // Next, build the image.
     const {imageId, stages} = await buildImageAsync(baseImageName, pathOrBuild, logResource, cacheFrom);
-    if (imageId === undefined) {
-        throw new Error("Internal error: docker build did not produce an imageId.");
-    }
```

Comment:
this can't happen.  `buildImageAsync` always throws, or returns a valid result with data set.  TS also validates this by ensuring that when we return a result that `imageId` is not-undefined.